src/basilisp/lang/vector.py

```diff
@@ -205,6 +216,30 @@ def rseq(self) -> ISeq[T]:
     def to_transient(self) -> TransientVector:
         return TransientVector(self._inner.evolver())
 
+    def reduce(self, f, init=reduce_init_obj):
+        if reduce_init_obj == init:
```

Comment:
```suggestion
        if init is _reduce_sentinel_obj:
```

For a sentinel, we can just use an identity check. I know `==` will do that for bare objects, but it's more explicit with less overhead to simply do it here.